### ISSUE TITLE
Fields marked as required using schema.path('x').required(true) are not marked as required.

### ISSUE THREAD

**Issue by makinde** [DScheglov/mongoose-schema-jsonschema#16]
If you have a schema, then add a field as being required programmatically, the resulting jsonschema doesn't mark the field as required.

**Comment by DScheglov** [comment]
@makinde Hi, sorry for late answer.

I've checked for mongoose v5.x:

```js
const mongoose = require('../index')();

const Schema = mongoose.Schema;

const BookSchema = new Schema({
  title: { type: String, required: true },
  year: Number,
  author: { type: Schema.Types.ObjectId, required: true, ref: 'Person' }
});

BookSchema.add({
	isbn: { type: String, required: true },
});

console.dir(BookSchema.jsonSchema(), { depth: null });
```

**Output**:
```js
({
  type: 'object',
  properties: {
    title: { type: 'string' },
    year: { type: 'number' },
    author: {
      type: 'string',
      'x-ref': 'Person',
      description: 'Refers to Person',
      pattern: '^[0-9a-fA-F]{24}$'
    },
    _id: { type: 'string', pattern: '^[0-9a-fA-F]{24}$' },
    isbn: { type: 'string' }
  },
  required: [ 'title', 'author', 'isbn' ]
})
```

**Comment by DScheglov** [comment]
Fixed in v1.3.0

**Comment by irvingv8** [comment]
Hello @DScheglov, I haven't dig too deep into this, however I don't think the current behaviour is correct:

I assume that when we say "programmatically required" we are talking about something like: `required: () => someLogic`

Then if a field is programmatically required, we can't tell at build time what will be the value of required, so I'd argue that it should default to the field being optional.

**Comment by DScheglov** [comment]
@irvingv8 , actually it makes sense.
So, if `required` is a function let's treat it as optional.

What do you think?

**Comment by DScheglov** [comment]
Fixed in `mongoose-schema-jsonschema@2.2.1`
